go/pkg/tree/tree_test.go

```diff
@@ -434,13 +436,54 @@ func TestComputeMerkleTreeErrors(t *testing.T) {
 			t.Fatalf("failed to construct input dir structure: %v", err)
 		}
 		t.Run(tc.desc, func(t *testing.T) {
-			if _, _, err := ComputeMerkleTree(root, tc.spec, chunker.DefaultChunkSize); err == nil {
+			if _, _, err := ComputeMerkleTree(root, tc.spec, chunker.DefaultChunkSize, &filemetadata.NoopFileMetadataCache{}); err == nil {
 				t.Errorf("ComputeMerkleTree(%v) succeeded, want error", tc.spec)
 			}
 		})
 	}
 }
 
+type stubFileMetadataCache struct {
+	Digest digest.Digest
+	Err    error
+}
+
+// Get implements the FileMetadataCache interface.
+func (s *stubFileMetadataCache) Get(_ string) (*filemetadata.Metadata, error) {
+	if s.Err != nil {
+		return nil, s.Err
+	}
+	return &filemetadata.Metadata{Digest: s.Digest}, nil
+}
+
+func TestComputeMerkleTreeUseCache(t *testing.T) {
```

Comment:
Can we have tests that actually exercise the cache while building the tree? We might need a fake cache for this, one that counts hits.